.gitignore

```diff
@@ -1,2 +1,6 @@
 .history/
-bin/josuke/josuke
+/conf/
+/work/
```

Comment:
`conf/` contains configuration files, the server JSON configuration for instance, and possibly the `cert.pem` and `key.pem` when TLS is enabled, files that must not be stored in Git because they are specific to running instances and/or may contain secrets.
`work/` is a working folder for the developer, files that would litter your project otherwise.

Reply:
To be precise, storing configuration in `conf/` is a convention I like to use, because you may independently store this configuration in a different Git, or use symbolic links, both, etc. you are free to do whatever you prefer, and you keep this configuration code as close as possible to the executable code.

Reply:
I'm not really keen on adding relatively meaningful directories in .gitignore.
`work/` directory I can agree, in case you want to store some random files for dev purpose.
But `conf/` I'm not sure. We're not supposed to store runtime config in git as you said. Storing real config in a project's directory and adding it in .gitignore feels dangerous. At some point .gitignore file could change, even by mistake, removing `conf/`, potentially making you push config without realizing it.

Reply:
I'm not really keen on adding relatively meaningful directories in .gitignore.
`work/` directory I can agree, in case you want to store some random files for dev purpose.
But `conf/` I'm not sure. We're not supposed to store runtime config in git as you said. Storing real config in a project's directory and adding it in .gitignore feels dangerous. At some point .gitignore file could change, even by mistake, removing `conf/`, potentially making you push config without realizing it.

Reply:
I use different accounts on Nix to clone the code and to run application instances. That way, with the right permissions, there is no chance that a private configuration might end up in the repository.

Reply:
Are you using this for running the exec only or does it have to do anything about dev or compilation ? I would not see a reason to add this directory in .gitignore if that's for the former